camera-chromecast-button.md

```diff
@@ -10,55 +10,101 @@ Casting to Chromecast requires the Chromecast and Scrypted Cloud plugins.
 ```ts
 class ChromecastCameraButton extends ScryptedDeviceBase implements OnOff, Settings {
     timeout: any;
+    readonly DEFAULT_TIMEOUT_SECS = 60;
 
     async turnOn() {
-        const chromecast = systemManager.getDeviceById<RTCSignalingClient & StartStop>(this.storage.getItem('chromecast'));
-        const camera = systemManager.getDeviceById<RTCSignalingChannel>(this.storage.getItem('camera'));
+        const ids = this.getJSON('chromecasts') as string[];
+        const camera = systemManager.getDeviceById<RTCSignalingChannel>(this.getJSON('camera') as string);
+        clearTimeout(this.timeout);
+
+        const promises = ids.map((id => {
+            return new Promise<void>(async (resolve, reject) => {
```

Comment:
Yeah, this whole promise construction and resolve/reject can be simplified by using what I suggested. I'll make the change myself since I'm not sure im explaining correctly.